File: app/src/main/java/dstudio/com/retrofitrxandroid/activity/BaseActivity.java
package dstudio.com.retrofitrxandroid.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.google.gson.Gson;

import butterknife.ButterKnife;
import dstudio.com.retrofitrxandroid.util.SharedPrefUtils;

/**
 * Created by janwelcris on 6/8/2017.
 */

public abstract class BaseActivity extends AppCompatActivity {
    protected Gson mGson;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResourceId());

        ButterKnife.bind(this);

        SharedPrefUtils.init(this);
        mGson = new Gson();

        onActivityCreated(savedInstanceState);

    }

    protected abstract int getLayoutResourceId();

    protected abstract void onActivityCreated(Bundle savedInstanceState);
}

File: app/src/main/java/dstudio/com/rxkotlinandroid/activity/BaseActivity.kt
package dstudio.com.rxkotlinandroid.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.google.gson.Gson

import dstudio.com.rxkotlinandroid.util.PreferenceWrapper

/**
 * Created by janwelcris on 7/12/2017.
 */
abstract class BaseActivity : AppCompatActivity() {


    protected var mPreferenceWrapper : PreferenceWrapper? = null
    protected var mGson: Gson? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mPreferenceWrapper = PreferenceWrapper.instanceOf(this)
        mGson = Gson()

    }


}
